R/validate.R

```diff
@@ -18,16 +18,84 @@
 #' @keywords check-forecasts
 #' @examples
 #' as_forecast(example_binary)
-#' as_forecast(example_quantile)
-as_forecast <- function(data, ...) {
+#' as_forecast(
+#'   example_quantile,
+#'   forecast_unit = c("model", "target_type", "target_end_date",
+#'                     "horizon", "location")
+#' )
+as_forecast <- function(data,
+                        ...) {
   UseMethod("as_forecast")
 }
 
 #' @rdname as_forecast
+#' @param observed (optional) Name of the column in `data` that contains the
+#' observed values. This column will be renamed to "observed".
+#' @param predicted (optional) Name of the column in `data` that contains the
+#' predicted values. This column will be renamed to "predicted".
+#' @param model (optional) Name of the column in `data` that contains the names
+#' of the models/forecasters that generated the predicted values.
+#' This column will be renamed to "model".
+#' @param forecast_unit (optional) Name of the columns in `data` (after
+#' renaming) that denote the unit of a single forecast.
+#' See [get_forecast_unit()] for details.
+#' @param quantile_level (optional) Name of the column in `data` that contains
+#' the quantile level of the predicted values. This column will be renamed to
+#' "quantile_level". Only applicable to quantile-based forecasts.
+#' @param sample_id (optional) Name of the column in `data` that contains the
+#' sample id. This column will be renamed to "sample_id". Only applicable to
+#' sample-based forecasts.
 #' @export
-as_forecast.default <- function(data, ...) {
+as_forecast.default <- function(data,
+                                observed = NULL,
+                                predicted = NULL,
+                                model = NULL,
+                                forecast_unit = NULL,
+                                quantile_level = NULL,
+                                sample_id = NULL,
+                                ...) {
```

Comment:
Ah those are good points!

Reply:
I think `as_forecast.default()` needs `...` because the generic has it and they always need to have the same arguments. But I like the idea of having a `forecast_type` argument

Reply:
Is that true for `...` args though? I'm not sure it is.

Reply:
Just checked, it's true for `...` args as well: 

```
checking S3 generic/method consistency (1.7s)
   as_forecast:
     function(data, ...)
   as_forecast.default:
     function(data, forecast_unit, forecast_type, observed, predicted,
              model, quantile_level, sample_id)
   See section ‘Generic functions and methods’ in the ‘Writing R
   Extensions’ manual.   
```

Reply:
Hm. Since additional arguments just have no effect at all, I would personally not throw an error for this. Given that the `...` are there I would not expect an error as a user if I provide an additional argument.